### ISSUE TITLE
Is the codebase currently runnable?

### ISSUE THREAD

**Issue by DingYX0731** [pnnl/chemreasoner#30]
I have noticed that the code has been modified these days. So I am wondering how I can run an example reasoning task. Should I run 
`run_automatically.py`? It seems that there are some bugs.
![QQ_1724174875335](https://github.com/user-attachments/assets/dbcaeed2-9dfe-4825-9b65-779f52085eac)

**Comment by HenrySprueill** [comment]
Yes, the codebase should be runnable. I've added some information to the README.md file under "Running the ICML Code". The file you'll want to use for reproducing the results of the ICML paper is ```src/scripts/run_icml_queries.py```

In the README are instructions on how to set up a .env file to connect to the azure OpenAI api, how to set up a local redis server to run GNN relaxations, and how to run the ICML code.

More recent modifications are related to future work or are minor tweaks to make the code simpler (such as batching and DataParallel)
